internal/sink/sink.go

```diff
@@ -338,14 +348,29 @@ func (k *Kafka) Update(id, topic string, retries int, servers []string, logger *
 	return nil
 }
 
-func (k Kafkas) Send(secret string, acceptType string, msg *wrp.Message) error {
-	//TODO: discuss with wes and john the default hashing logic
-	//for now: when no hash is given we will just loop through all the kafkas
+func (k KafkaSink) Send(secret string, acceptType string, msg *wrp.Message) error {
 	var errs error
-	for _, kafka := range k {
-		err := kafka.send(secret, acceptType, msg)
-		if err != nil {
-			errs = errors.Join(errs, err)
+	if k.HashField != "" {
+		key := GetKey(k.HashField, msg)
+		for i, kafka := range k.Kafkas {
```

Comment:
I like what you did here, but I think we can make this easier to maintain and read 🙂 

tl;dr i would suggest to leverage the fact you're using hashes by converting `k.Kafkas` to a map and then using `k.Hash.GetServer(key)` directly, e.g.: `err := k.Kafkas[GetKey(k.HashField, msg)].send(secret, acceptType, msg)`

that would simplify the code a bit